js/src/components/contact-information/phone-number-card.js

```diff
@@ -105,67 +128,114 @@ function PhoneNumberContent( {
+				onPhoneNumberChange={ onPhoneNumberChange }
+			/>
+		</>
+	) : (
+		<AppSpinner />
+	);
+
+	return (
+		<AccountCard
+			{ ...basePhoneNumberCardProps }
+			description={ __(
+				'Please enter a phone number to be used for verification.',
+				'google-listings-and-ads'
+			) }
+		>
+			{ phoneNumberContent }
+		</AccountCard>
+	);
+}
+
+/**
+ * Renders phone number data in Card UI and is able to edit.
+ *
+ * @param {Object} props React props.
+ * @param {PhoneNumber} props.phoneNumber Phone number data.
+ * @param {boolean} [props.isPreview=false] Whether to display as preview UI.
+ * @param {boolean|null} [props.initEditing=null] Specify the inital UI state. This prop would be ignored if `isPreview` is true.
+ *     `true`: initialize with the editing UI.
+ *     `false`: initialize with the viewing UI.
+ *     `null`: determine the initial UI state according to the `data.isValid` after the `phoneNumber` loaded.
+ * @param {Function} [props.onEditClick] Called when clicking on "Edit" button.
+ *     If this callback is omitted, it will enter edit mode when clicking on "Edit" button.
+ * @param {onPhoneNumberChange} [props.onPhoneNumberChange] Called when inputs of phone number are changed in edit mode.
+ */
 export default function PhoneNumberCard( {
 	phoneNumber,
-	isPreview,
-	initEditing,
+	isPreview = false,
+	initEditing = null,
 	onEditClick,
 	onPhoneNumberChange = noop,
 } ) {
-	const [ isEditing, setEditing ] = useState( initEditing );
 	const { loaded, data } = phoneNumber;
-
-	const editButton = isEditing ? null : (
-		<AppButton
-			isSecondary
-			onClick={ () => {
-				if ( onEditClick ) {
-					onEditClick();
-				} else {
-					setEditing( true );
-				}
-			} }
-		>
-			{ __( 'Edit', 'google-listings-and-ads' ) }
-		</AppButton>
+	const [ isEditing, setEditing ] = useState(
+		isPreview ? false : initEditing
 	);
 
-	let description = null;
-	let phoneNumberContent = null;
+	useEffect( () => {
+		if ( loaded && isEditing === null ) {
+			setEditing( ! data.isValid );
+		}
+	}, [ loaded, data.isValid, isEditing ] );
+
+	// Return a simple loading AccountCard since the initial edit state is unknown before loaded.
+	if ( isEditing === null ) {
+		return (
+			<AccountCard
+				{ ...basePhoneNumberCardProps }
+				indicator={ <Spinner /> }
+			/>
+		);
+	}
 
 	if ( isEditing ) {
-		description = __(
-			'Please enter a phone number to be used for verification.',
-			'google-listings-and-ads'
+		return (
+			<EditPhoneNumberCard
+				phoneNumber={ phoneNumber }
+				onPhoneNumberChange={ onPhoneNumberChange }
+			/>
 		);
+	}
 
-		if ( loaded ) {
-			phoneNumberContent = (
-				<>
-					<CardDivider />
-					<PhoneNumberContent
-						initCountry={ data.country }
-						initNationalNumber={ data.nationalNumber }
-						onPhoneNumberChange={ onPhoneNumberChange }
-					/>
-				</>
-			);
-		} else {
-			phoneNumberContent = <AppSpinner />;
-		}
+	let description = null;
+	let indicator = null;
```

Comment:
💅  I'd specify `Spinner` as the default indicator so that we don't need the `else` block below.

```suggestion
	let indicator = <Spinner />;
```